addons/fleet/models/fleet_vehicle.py

```diff
@@ -0,0 +1,549 @@
+                                     help='Unit of the odometer',
+                                     required=True, default='kilometers')
+    transmission = fields.Selection([('manual', 'Manual'), ('automatic', 'Automatic')],
+                                    help='Transmission Used by the vehicle')
+    fuel_type = fields.Selection([('gasoline', 'Gasoline'), ('diesel', 'Diesel'),
+                                            ('electric', 'Electric'), ('hybrid', 'Hybrid')],
+                                 help='Fuel Used by the vehicle')
+    horsepower = fields.Integer()
+    horsepower_tax = fields.Float(string='Horsepower Taxation')
+    power = fields.Integer(help='Power in kW of the vehicle')
+    co2 = fields.Float(string='CO2 Emissions', help='CO2 emissions of the vehicle')
+    image = fields.Binary(related='model_id.make_id.image', string="Logo", store=True)
+    image_medium = fields.Binary(related='model_id.make_id.image_medium', string="Logo (medium)", store=True)
+    image_small = fields.Binary(related='model_id.make_id.image_small', string="Logo (small)", store=True)
+    contract_renewal_due_soon = fields.Boolean(compute='_compute_get_contract_reminder',
+                                               search='_search_contract_renewal_due_soon',
+                                               string='Has Contracts to renew')
+    contract_renewal_overdue = fields.Boolean(compute='_compute_get_contract_reminder',
+                                              search='_search_get_overdue_contract_reminder',
+                                              string='Has Contracts Overdue')
+    contract_renewal_name = fields.Text(compute='_compute_get_contract_reminder', string='Name of contract to renew soon')
+    contract_renewal_total = fields.Integer(compute='_compute_get_contract_reminder',
+                                            string='Total of contracts due or overdue minus one')
+    car_value = fields.Float(help='Value of the bought vehicle')
+
+    _sql_constraints = [('unique_chassis_number', 'unique(vin_sn)', 'Same Chassis Number is already exists.')]
+
+    @api.one
+    @api.depends('license_plate')
+    def _compute_vehicle_name(self):
+        self.name = ('%s' + ' / ' + '%s' + ' / ' + '%s') % (self.model_id.make_id.name, self.model_id.name, self.license_plate)
+
+    @api.multi
+    def _compute_get_contract_reminder(self):
+        LogContract = self.env['fleet.vehicle.log.contract']
+        current_date = fields.Datetime.from_string(fields.Date.context_today(self))
+        for vehicle in self:
+            overdue = False
+            due_soon = False
+            total = 0
+            name = ''
+            for contract in LogContract.search([('vehicle_id', '=', vehicle.id), ('state', 'in', ('open', 'toclose')), ('expiration_date', '!=', None)],
+                                                limit=1, order='expiration_date asc'):
```

Comment:
you may avoid a call to search(...) by replacing it with,

if vehicle.log_contracts_ids:
    contract = vehicle.log_contracts_ids.filtered(lambda contract:
                         contract.state in ('open', 'toclose') \
                         and contract.expiration_date != None).sorted(lambda contract: \
                         contract.expiration_date)[0]